src/main/java/chessmaster/storage/Storage.java

```diff
@@ -196,6 +210,46 @@ public ChessTile[][] loadBoard() throws ChessMasterException {
         return boardTiles;
     }
 
+    public void executeSavedMoves(Color playerColor,
+                                  ChessBoard board,
+                                  ChessTile[][] otherBoard,
+                                  Human human,
+                                  CPU cpu) throws ChessMasterException {
+        ArrayList moveStringList = new ArrayList<String>();
+        ArrayList<String> humanMoves = loadHumanMoves();
+        ArrayList<String> cpuMoves = loadCPUMoves();
+
+        // Merge move string arrays into a singular array
+        if (playerColor.isWhite()) {
+            for (String move : humanMoves) {
+                moveStringList.add(move);
+            }
+
+            for (int i = 0; i < cpuMoves.size(); i ++) {
+                moveStringList.add(2 * i + 1, cpuMoves.get(i));
+            }
+        } else if (playerColor.isBlack()) {
+            for (String move : cpuMoves) {
+                moveStringList.add(move);
+            }
+
+            for (int i = 0; i < humanMoves.size(); i ++) {
+                moveStringList.add(2 * i + 1, humanMoves.get(i));
+            }
+        } else {
+            throw new LoadBoardException();
+        }
+
+        // Execute move string array
+        board.executeMoveArray(moveStringList, human, cpu);
```

Comment:
Create the new board local to the `executeSavedMoves` function, not from the `ChessMaster` class. 

This is the same comment as the one in ChessMaster, just a reminder

Reply:
I was planning on creating the board to be played on based on the moves, this is correct. The board loaded from the file (in the form of ChessTile[][]) is used to check the validity of the current board after it has been loaded.